HNG_model.py

```diff
@@ -87,9 +96,26 @@ def residual(t, solution):
 x = [ran.random()*r_range for i in range(nucleii)]
 y = [ran.random()*r_range for i in range(nucleii)]
 
+#%%
+with PdfPages('output' +  dt.datetime.now().strftime("%Y%m%d") + '.pdf') as pdf:
+    plt.figure(0)
+    plt.plot(t,radius)
+    pdf.savefig()
+    plt.close()
+
+    plt.figure(1)
+    plt.plot(t,concentrations)
+    pdf.savefig()
+    plt.close()
+
+    for i in range(0,len(t), int(0.01*max(t))):
```

Comment:
Can you add a short comment to describe what this is doing?
To my eye, you are creating 100 different figures, each with a scatter plot somehow related to the particle radius distribution at time `t`?  The `r_range` calculation above, plus the `s=np.ones_like(x)*3000*radius[I]` calculation below aren't immediately intuitive to me.  I'll download and run the code to see if I can grasp it, but either way, a bit of documentation here would be generally helpful.

Reply:
I thought it would be nice to visualize the particle growth.  I'm sure there's a better way to do this, but the 0.01*max(t) is basically saying that 10% of the time you should generate plots so for a time of 200 s would be 20 plots.   Right now it randomly plots the nuclei, and then the radius of the points grows.  The 3000 is actually a meaningless number that happened to make all of the points visible.  It's not very well thought out, but it was more of something I did just to experiment and learn I guess.